Snake.fs

```diff
@@ -53,44 +62,34 @@ module SnakeGame =
             | [] -> carry
         zip Tail (List.rev correctLength)
 
-    module Graphics = 
+    /// 🌍 Separate world logic from rendering
+    module World =
         let private random = new Random()
-        let private head = "🤢"
-        let private belly = "🟢"
-        let private display = "⬜"
-        let private errorDisplay = "🟥"
-
-        let private food = [|"🐁";"🐀";"🐥";"🪺";"🐸";"🐛";"🪰";"🐞";"🦗"|]
-        let private randomFood () = food.[random.Next(food.Length - 1)]
-
-        let isFood square = Array.contains square food
-        let isFreeSpace square = square = display || square = errorDisplay
-        let isOccupied square = 
-            match square with 
-            | square when isFreeSpace square -> false 
+        let isFood square =
+            match square with
+            | CellFood(_) -> true
+            | _ -> false
```

Comment:
I think at the time I couldn't work out how to compare against a parameterised union without `match`...

(I still don't know how you do that in F#, but that's why this helper func is uggs)